App Secrets, Attempt no.3 (#250)
Esri/data-collection-ios: README.md

#### 3. Configure app secrets

As a best-practices principle, the project conceals app secrets from source code by generating and compiling an `AppSecrets.swift` source code file at build time using a custom build rule.

This build rule looks for a secrets file stored in the project directory, `$(PROJECT_DIR)/.secrets`.

1. Create a hidden secrets file in the project's root directory.

  ```bash
  touch data-collection/.secrets
  ```

2. Add your registered app's **Client ID** to the secrets file.

  ```bash
  echo ARCGIS_CLIENT_ID=your-client-id >> data-collection/.secrets
  ```

  > Replace 'your-client-id' with your app's **Client ID**.

3. _Optionally_ add your **License Key** to the secrets file. Licensing the app will remove the 'Licensed for Development' watermark. Licensing the app is optional in development but required for production. Aquire your license key from your [dashboard](https://developers.arcgis.com/dashboard).

  ```bash
  echo ARCGIS_LICENSE_KEY=your-license-key >> data-collection/.secrets
  ```

  > Replace 'your-license-key' with your **License Key**.

> A Lite license is free and provides developers a good starting point. Visit the developers website to learn more about [licensing your ArcGIS Runtime app](https://developers.arcgis.com/pricing/licensing/).

Reviewer: ```suggestion
A Lite license is free and enables all of the capabilities required by this app. Visit the developer's website to learn more about [licensing your ArcGIS Runtime app](https://developers.arcgis.com/pricing/licensing/).
```
